x-pack/solutions/security/packages/kbn-securitysolution-utils/src/time_duration/to_largest_time_duration.test.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { toLargestTimeDuration } from './to_largest_time_duration';
+
+describe('toLargestTimeDuration', () => {
+  it.each([
+    [10, '10ms'],
+    [5000, '5s'],
+    [600000, '10m'],
+    [25200000, '7h'],
+    [777600000, '9d'],
+    [5184000000, '60d'],
+    [-20, '-20ms'],
+    [-3000, '-3s'],
+    [-300000, '-5m'],
+    [-18000000, '-5h'],
+    [-604800000, '-7d'],
+    [-10368000000, '-120d'],
+  ])('converts "%s" to the largest time duration', (duration, expected) => {
+    const result = toLargestTimeDuration(duration);
+
+    expect(result).toBe(expected);
+  });
```

Comment:
Tests are missing for inputs that are not equal to whole seconds, minutes, hours and days.